src/djimatrice_drone.py

```diff
@@ -0,0 +1,230 @@
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
+
+    def resume_mission(self):
+        try:
+            print("Attempting to resume drone mission...")
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/mission_waypoint_action', 'dji_sdk/MissionWpAction')
+            request = roslibpy.ServiceRequest({"action": Drone.WaypointActions.RESUME})
+
+            print('Calling mission_waypoint_action resume service...')
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Mission failed to resume"}
+        # TODO: Upon failure, revert back to original setting
+        return result
+    
+    def resume_mission_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
+
+    # TODO Implement
+    def land_drone(self):
+        try:
+            print("Attempting to call drone specific service...")
+            #TODO change to actual service call and type
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/drone_task_control', 'dji_sdk/DroneTaskControl')
+            # TODO check service type on drone aka check if 6 is correct
+            request = roslibpy.ServiceRequest({"task": DroneTaskControl.LAND})
+
+            print('Calling land_drone service...')
+            #TODO parse service.call(request)
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Drone landing failed"}
+        # self.land_drone_callback(result)
+        return result
+    
+    def land_drone_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
+
+    # TODO Implement
+    def fly_home(self):
+        try:
+            print("Attempting to call drone specific service...")
+            #TODO change to actual service call and type
+            service = roslibpy.Service(self.ROS_master_connection, 'dji_sdk/drone_task_control', 'dji_sdk/DroneTaskControl')
+            # TODO check service type on drone aka check if 1 is correct
+            request = roslibpy.ServiceRequest({"task": DroneTaskControl.GO_HOME})
+
+            print('Calling fly_home service...')
+            #TODO parse service.call(request)
+            result = service.call(request)
+            print('Service response: {}'.format(result))
+        except:
+            result = {"success":False, "message":"Drone flying home failed"}
+        return result
+
+    def fly_home_drone_callback(self, result):
+        # TODO: Add more after figuring out what callback is used to update
+        return result["success"]
+
+    #TODO Implement
+    def update_mission_helper(self, action):
```

Comment:
Why do we have a separate update mission?

Reply:
I think we should keep the helper function b/c it specifies the result of each action. Also, it is called multiple times, so I think it's good to keep one central function.